### PR TITLE
Allow `DEBUG2`-logging in sub-protocols + nitpick debug messages + various

### PR DESCRIPTION
**Pull request by veox**
### What was wrong?

* Sub-protocols couldn't use `DEBUG2` logging level;
* `DEBUG`-level logging was a bit too noisy, and displaying `int`/`bytes` where hex-strings make more sense;
* --help has not been updated for the TRACE -> DEBUG2 rename.

### How was it fixed?

Major:

* ETH and BCC sub-protocols now also derive from `HasExtendedDebugLogger`;
* `ETH/Status` and `BCC/Status` sub-protocol-level logging was moved to level `DEBUG2`.

Minor:

* A few debug messages now have `to_hex()`;
* `trinity --help` output for log levels now generated on-the-fly;

Extra:

* peer (IP) is printed in generic "Disconnecting" message;
* update date in `LICENSE` file.

Detailed reasoning is in commit messages.

### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: p2p/peer.py [hunk 1]
```diff
@@ -578,7 +578,7 @@ def _disconnect(self, reason: DisconnectReason) -> None:
             raise ValueError(
                 f"Reason must be an item of DisconnectReason, got {reason}"
             )
-        self.logger.debug("Disconnecting from remote peer; reason: %s", reason.name)
+        self.logger.debug(f"Disconnecting from remote peer {self.remote}; reason: {reason.name}")
```
Comment (pipermerriam) [inline p2p/peer.py]: This should be converted back to `%` style string formatting.

https://github.com/ethereum/snake-charmers-tactical-manual/blob/master/style-guide.md#formatting-log-messages

File: trinity/protocol/eth/proto.py [hunk 2]
```diff
@@ -62,7 +60,6 @@ def send_handshake(self, chain_info: ChainInfo) -> None:
             'genesis_hash': chain_info.genesis_hash,
         }
         cmd = Status(self.cmd_id_offset)
-        self.logger.debug("Sending ETH/Status msg: %s", resp)
```
Comment (pipermerriam) [inline trinity/protocol/eth/proto.py]: Rather than remove why not move to `TRACE` level?

File: p2p/peer.py (hunk 1 above)
Comment (veox) [reply p2p/peer.py]: Right! Will do tomorrow - will check other places where `f"{stuff}"` was introduced, too.

File: trinity/protocol/eth/proto.py (hunk 2 above)
Comment (veox) [reply trinity/protocol/eth/proto.py]: I tried logging at `debug2` in https://github.com/ethereum/trinity/pull/113/commits/25990d86a7f3345ff4e18a46c02410b7c5686a74, but then https://github.com/ethereum/trinity/pull/113/commits/3ad83509b3bdab2392d120b662adc2efaaea8e12 happened.

In all honesty, I got agitated. Could review `Protocol` to have `debug2` included instead.

Comment (pipermerriam) [reply trinity/protocol/eth/proto.py]: > Could review Protocol to have `debug2` included in its `Logger` instead.

I'm :+1: on this route.

Comment (veox) [reply trinity/protocol/eth/proto.py]: This might actually be an issue in `mypy`:

https://github.com/python/mypy/issues/5973

Comment (veox) [reply trinity/protocol/eth/proto.py]: `trinity/proto` sub-protocols should now be able to use `logger.debug()`.

**Comment by veox** [comment]
@jannikluhn @hwwhww @ChihChengLiang I've taken the liberty of touching BCC (not just ETH) sub-protocol in this PR, in commit f46efa8cb9671736643e4eb86fc98739a67b0b63. (For ETH, it was 363bb8c20975e0cb9a21b2320ec9f11695c5cdcf.)

The `"Sending BCC/Status msg"` log line will now appear on level `DEBUG2` (instead of previous `DEBUG`), and from logger `trinity.protocol.eth.proto.BCCProtocol` (i.o.p. `p2p.bcc.BCCProtocol`).

Do ping if this change is undesired, it's (still) easily-revertable.

**Comment by veox** [comment]
@pipermerriam I think this change - allowing `logger.debug2()` - makes the PR unsquashable. It'd make unrelated changes inseparable. :(

**Pull request merged**